security/fma-shared-lockbox.md

```diff
@@ -0,0 +1,147 @@
+- **Detection:** Verify deployed bytecode against expected hash values before proxy to target the implementation contract.
+- **Recovery Path(s):** Pause the system if a Solidity compiler bug is identified.
+
+### FM5: Buggy upgrade over the `SharedLockbox`
+
+- **Description:** The `SharedLockbox` is a proxied contract, which means its implementation can be upgraded. While this provides flexibility for fixes and improvements, a buggy or malicious upgrade could introduce vulnerabilities that compromise all ETH liquidity.
+- **Risk Assessment:** High.
+    - Potential Impact: Critical. If an upgrade introduces a bug or an exploitable backdoor, all ETH stored in the `SharedLockbox` could be drained.
+    - Likelihood: Low to Medium. Upgrades are controlled by Security Council, after governance approval. However, upgrade mistakes or compromised keys remain a risk.
+- **Mitigations:** Any new implementation must go through the audit and testing. Upgrades should not be regular in this contract and it should maintain minimal code. Upgrade procedures and keys must follows the proper security practices.
+- **Detection:** Verify proxy and implementation contracts
+- **Recovery Path(s):** Pause the system through `SuperchainConfig` to halt the `SharedLockbox`.
+
+### FM6: Unauthorized use of `DEPOSITOR_ACCOUNT` to call `addDepedency` in `DependencyManager`
+
+- **Description:** On L2, only the `DEPOSITOR_ACCOUNT` can call `addDependency` in `DependencyManager`. If attackers can spoof or impersonate that account, they can push arbitrary additions to the dependency set, eventually letting malicious portals into the `SharedLockbox`.
+- **Risk Assessment:** Medium.
+    - Potential Impact: High. Attackers could create fake or malicious chain entries that, if get accepted on L1, would authorize an attacker-controlled portal in `SharedLockbox`.
+    - Likelihood: Low. The `DEPOSITOR_ACCOUNT` is typically used by the derivation pipeline or otherwise restricted. Previously, there were specific deposit calls that used this account, but those are no longer in use. If the `DEPOSITOR_ACCOUNT` becomes callable in other ways under future upgrades, the attack surface would increase.
+- **Mitigations:** There should be tests that ensure `DEPOSITOR_ACCOUNT` only can be called as part of a coordinated upgrade. Review `DEPOSITOR_ACCOUNT` privileges if it is used for other features in the future. Ensure interop fault proof is properly set to reject states that add a non-expected chain.
+- **Detection:** Monitor `DependencyAdded` events to catch unexpected additions. Do periodical checks to `dependencySet` to review the existing dependency set recorded in the chain.
+- **Recovery Path(s):** Because removing a chain from the dependency set is currently irreversible, a hard fork will be the only way to revert an unauthorized addition. Patch any potential issue around `DEPOSITOR_ACCOUNT`.
+
+### FM7: Unable to add new Chains by Reaching the `uint8` limit in `addDependency`
```

Comment:
Why limit it to uint8? Can we just make it a uint64 or uint256 and remove this failure mode?

Reply:
The original spec passed the dep set via L1 attributes, so it was limited to u8 to keep it small enough to pass on every block. The max size should actually be smaller, given it should not be possible to create an unprovable cluster